src/main/java/use_cases/GameState.java

```diff
@@ -98,4 +97,14 @@ public boolean getHasWinner() {
     public HashMap<Player, Integer> getPlayersAndCards() {
         return playersAndCards;
     }
+
+    private HashMap<Player, Integer> updatePlayersAndCards(ArrayList<Player> players) {
+        HashMap<Player, Integer> playersAndCards = new HashMap<>();
```

Comment:
This is already initialized at the top, see line 19, is it necessary to create a new HashMap again?

Reply:
its returning a local variable that happens to have the same name, ill change it so its more clear